compiler/src/iree/compiler/Codegen/Common/GPU/VectorLayoutProvider.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_CODEGEN_VECTOR_LAYOUT_PROVIDER_H
+#define IREE_COMPILER_CODEGEN_VECTOR_LAYOUT_PROVIDER_H
+
+#include <array>
+#include "iree-dialects/Dialect/VectorExt/IR/VectorExtOps.h"
+#include "iree/compiler/Codegen/Common/VectorLayoutAnalysis.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/GPU/IR/GPUDialect.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/IR/AffineMap.h"
+
+namespace mlir::iree_compiler {
+
+class LayoutProvider {
```

Comment:
Could you add some comments about the classes being added and heirarchy? Some questions that pop up immediately is the layout is completely agnostic about distribution, so have methods like `getDistributedShape` is confusing.